### DISCUSSION TITLE
When the circle is zoomed in, the uneven edges can be clearly seen.

### DISCUSSION THREAD

**Discussion by Tangkkk** [pixijs/pixijs#7372] [Q&A]
Is there a parameter to set the smoothness of the circle?
But not anti-aliasing

**Comment by bigtimebuddy** [comment]
In general, Graphic drawing in PixiJS is adaptive. All shapes (circle, rectangles, etc) are converted to triangles which means that we don't want to render _too many_ triangle or that might make your application slow. This means: the initial radius of your circle matters in terms of figuring out how many triangles to generate. 

Here's an example of this: https://jsfiddle.net/bigtimebuddy/g7n0foL2/

Both circles are the same size, but one is draw with a smaller radius and scaled up. Here you can see the triangles as edges along the edge that should be smooth The other circle is drawn at 100% size.

**Comment by ShukantPal** [comment]
You can try using @pixi-essentials/conic - it will parametrically render your circle in the shader and doesn't use tessellation for approximating the circle.

**Comment by pitkes22** [comment]
A simple solution for this is to render the circle larger than needed and then scale it down. This way, it will retain details as if it were large but scaled down. Essentially, if you render it initially at the largest size it will ever need when zoomed in, it will never lose any details when zoomed in.

```ts
const SCALE = 100
graphics.scale.set(1/SCALE)
graphics.circle(x * SCALE, y * SCALE, 3 * SCALE)
graphics.fill('red')
```

**Reply by ShukantPal** [reply]
How do you know what SCALE is a priori?

**Reply by pitkes22** [reply]
If you have a maximum zoom level set, it’s straightforward: render the element scaled as if it were at the maximum zoom. If the zoom level is unrestricted, you can use the current zoom level, ensuring the element is always rendered at the exact size it appears on screen.

The issue is that if you render a circle with a radius of `r=10` in Pixi, it assumes the circle will only be `10px` on the screen. When you then scale up the entire viewport with zoom, the missing details become noticeable. My solution simply renders the circle at a larger size and then scales it down. This way, the element still has a radius of `r=10` but retains details as if it were larger. By matching the scale to the zoom level, elements will have details corresponding to their actual size on screen.

**Reply by ShukantPal** [reply]
Oh, I misread your comment as a suggestion for PixiJS. Yes, the code author can easily render at a larger radius with opposite scaling.

**Comment by shirecoding** [comment]
i think you need to account for the parent's scale in general when you draw graphics, else it will look blurry.

```ts
const scale = {x:1.2, y:1.2} // you should calculate this
circle.scale.x = scale/parent.scale.x
circle.scale.y = scale/parent.scale.y
```
